swift/Samples/ModalContainer/Sources/ModalContainerViewController.swift

```diff
@@ -0,0 +1,325 @@
+
+        setNeedsStatusBarAppearanceUpdate()
+
+        if #available(iOS 11.0, *) {
+            setNeedsUpdateOfHomeIndicatorAutoHidden()
+            setNeedsUpdateOfScreenEdgesDeferringSystemGestures()
+        }
+
+        // Set the topmost screen to be the accessibility modal
+        presentedScreens.last?.viewController.view.accessibilityViewIsModal = true
+        for modal in presentedScreens.dropLast() {
+            modal.viewController.view.accessibilityViewIsModal = false
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        addChild(baseScreenViewController)
+        view.addSubview(baseScreenViewController.view)
+        baseScreenViewController.didMove(toParent: self)
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        baseScreenViewController.view.frame = view.bounds
+
+        presentedScreens.forEach {
+            let displayInfo = ModalDisplayInfo(containerSize: view.bounds.size, style: $0.style, animated: $0.animated)
+            $0.viewController.view.frame = displayInfo.frame
+            $0.dimmingView?.frame = view.bounds
+        }
+    }
+
+    override var childForStatusBarStyle: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    override var childForStatusBarHidden: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    override var childForHomeIndicatorAutoHidden: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    override var childForScreenEdgesDeferringSystemGestures: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    public override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return topmostScreenViewController?.supportedInterfaceOrientations ?? super.supportedInterfaceOrientations
+    }
+}
+
+fileprivate struct ModallyPresentedScreen<ModalScreen: Screen> {
+    var viewController: DescribedViewController
+    var style: ModalContainerScreen<ModalScreen>.Modal.Style
+    var key: String
+    var dimmingView: UIView?
+    var animated: Bool
+
+    var identifier: ModalIdentifier<ModalScreen> {
+        return ModalIdentifier(
+            style: style,
+            key: key,
+            animated: animated
+        )
+    }
+}
+
+extension ModalContainerScreen.Modal {
+    fileprivate var identifier: ModalIdentifier<BaseScreen> {
+        return ModalIdentifier(
+            style: style,
+            key: key,
+            animated: animated
+        )
+    }
+}
+
+fileprivate struct ModalIdentifier<ModalScreen: Screen>: Hashable {
+    var style: ModalContainerScreen<ModalScreen>.Modal.Style
+    var key: String
+    var animated: Bool
+
+    static func == (lhs: ModalIdentifier, rhs: ModalIdentifier) -> Bool {
+        return lhs.style == rhs.style
+            && lhs.key == rhs.key
+            && lhs.animated == rhs.animated
+    }
+
+    func hash(into hasher: inout Hasher) {
```

Comment:
We can probably have this be synthesized.

Reply:
done